components/UserObserver.php

```diff
@@ -44,11 +44,20 @@ public function updateUser($params)
 		if (in_array($params['username'],Yii::app()->params['local_users'])) {
 			return;
 		}
-
-		if (Yii::app()->params['mehstaffdb_always_refresh'] || $this->isStale($params['username'], $params['institution_authentication_id'])) {
+		if (!isset($params['institution_authentication_id'])) {
+			$institutionId = Institution::model()->find('remote_id = "'.  Yii::app()->params['institution_code'] . '"')->id;
+			$institution_authentication_id = InstitutionAuthentication::model()->find('institution_id="'. $institutionId .'"')->id;
+		} else {
+			$institution_authentication_id = $params['institution_authentication_id'];
+		}
+			 
+		if (Yii::app()->params['mehstaffdb_always_refresh'] || $this->isStale($params['username'], $institution_authentication_id)) {
 			try {
 				$username = $params['username'];
-				$institution_authentication_id = $params['institution_authentication_id'];
+				if(!$institution_authentication_id) {
+					$institutionId = Institution::model()->find('remote_id = "'.  Yii::app()->params['institution_code'] . '"')->id;
+					$institution_authentication_id = InstitutionAuthentication::model()->find('institution_id="'. $institutionId .'"')->id;
+				}	
```

Comment:
1. it probably will be better to use parameter in `model()->find()`, like `model()->find('condition = :condition', array(':condition' => $value))`
2. line 57 to 60, the `if(!$institution_authentication_id)` feels a bit redundant. Should the `$institution_authentication_id` 100% be available at this stage? If not, should the situation be handled at outer if statement (`if (Yii::app()->params['mehstaffdb_always_refresh'] || $this->isStale($params['username'], $institution_authentication_id))`)?

Reply:
1. you are right, I fixed it.
2. I remove lines 57-60, it's not needed there